src/group_theory/perm/cycles.lean

```diff
@@ -446,6 +493,63 @@ quotient.rec_on_subsingleton (@univ α _).1
   (λ l h, trunc.mk (cycle_factors_aux l f h))
   (show ∀ x, f x ≠ x → x ∈ (@univ α _).1, from λ _ _, mem_univ _)
 
+section cycle_factors_finset
+
+variables [fintype α] (f : perm α)
+
+/-- Factors a permutation `f` into a `finset` of disjoint cyclic permutations that multiply to `f`.
+-/
+def cycle_factors_finset [fintype α] (f : perm α) : finset (perm α) :=
+(trunc_cycle_factors f).lift
+  (λ (l : {l : list (perm α) // l.prod = f ∧ (∀ g ∈ l, is_cycle g) ∧ l.pairwise disjoint}),
+    l.val.to_finset) (λ ⟨l, hl⟩ ⟨l', hl'⟩, list.to_finset_eq_of_perm _ _
+      (list_cycles_perm_list_cycles (hl'.left.symm ▸ hl.left) hl.right.left (hl'.right.left)
+        hl.right.right hl'.right.right))
+
+/-- The product of cycle factors is equal to the original `f : perm α`. We operate in the
+`list (perm α)` because `finset.prod` requires the multiplication across all `x : perm α`
+to be commutative. -/
+lemma cycle_factors_finset_prod (l : list (perm α)) (h : l.nodup)
+  (hl : l.to_finset = cycle_factors_finset f) : l.prod = f :=
+begin
+  obtain ⟨x, hx⟩ := trunc.exists_rep f.trunc_cycle_factors,
+  have ht : cycle_factors_finset f = x.val.to_finset,
+  { rw [cycle_factors_finset, ←hx, trunc.lift_mk] },
+  have hp : l ~ x,
+  { rw list.perm_ext h,
+    { simp [←list.mem_to_finset, hl, ht] },
+    { refine nodup_of_pairwise_disjoint _ x.prop.right.right,
+      intro hm,
+      simpa using (x.prop.right.left _ hm).ne_one } },
+  rw ←x.prop.left,
+  refine (hp.symm.prod_eq' _).symm,
+  refine x.prop.right.right.imp _,
+  exact λ _ _, disjoint.mul_comm
+end
+
+lemma cycle_factors_finset_mem_is_cycle (p : perm α) (hp : p ∈ cycle_factors_finset f) :
+  p.is_cycle :=
+begin
+  obtain ⟨x, hx⟩ := trunc.exists_rep f.trunc_cycle_factors,
+  have ht : cycle_factors_finset f = x.val.to_finset,
+  { rw [cycle_factors_finset, ←hx, trunc.lift_mk] },
+  rw [ht, list.mem_to_finset] at hp,
+  exact x.prop.right.left _ hp
+end
+
+lemma cycle_factors_finset_pairwise_disjoint_of_ne ⦃p : perm α⦄ (hp : p ∈ cycle_factors_finset f)
```

Comment:
"pairwise_disjoint_of_ne" sounds redundant to me. Would "pairwise_disjoint" or "disjoint_of_ne" be better?